gfx-memory/src/allocator/linear.rs

```diff
@@ -150,42 +177,43 @@ impl<B: Backend> LinearAllocator<B> {
         self.line_size / 2
     }
 
-    fn cleanup(&mut self, device: &B::Device, off: usize) -> Size {
+    fn cleanup(&mut self, device: &B::Device, free_memory: bool) -> Size {
         let mut freed = 0;
-        while self.lines.len() > off {
+        for _ in 0..self.lines.len() {
             if self.lines[0].allocated > self.lines[0].freed {
                 break;
             }
 
             let line = self.lines.pop_front().unwrap();
             self.finished_lines_count += 1;
 
-            match Arc::try_unwrap(line.memory) {
-                Ok(mem) => unsafe {
-                    log::trace!("Freed 'Line' of size of {}", mem.size());
-                    if mem.is_mappable() {
-                        device.unmap_memory(mem.raw());
+            if Arc::strong_count(&line.memory) == 1 {
```

Comment:
perhaps we can move this check under the `else {..}` branch, since the `if {..}` calls `free_memory`, which also checks for the very same thing?

Reply:
yes, good idea.